test/CMakeLists.txt

```diff
@@ -81,7 +81,7 @@ if(MSVC)
     endif()
 else()
     # Non-MSVC specific configuration (original content)
-    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD 23)
```

Comment:
This is not right.

Reply:
This is what I mentioned on the other PR, I think we will need to use at least C++ 20 to gain access to `constexpr` `std::array`. Maybe you want to have a look at the `std::array` component of this? I'll be signing off right now since is very late here in the UK!

Reply:
I don't understand what you mean, `std::array` has been `constexpr` from the start in '11

Reply:
this change is outdated.